### PR TITLE
Find with prefix

### PR DESCRIPTION
**Pull request by zhengweii**
To add test cases by sat (done)
To fix code style errors once tests have been pushed (done)
To update the description for find under help command as well (done)

Ready for review

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -177,19 +177,21 @@ the warehouse at index 1 of the list of warehouses.
...
 
-Format: `find TYPE KEYWORD [KEYWORD]…​`
+Format: `find ct/TYPE [n/NAME…​] [pd/PRODUCT_NAME…​] [r/REMARK…​]`
 
-* KEYWORD specified is case-insensitive and must contain at least one keyword.
-* The TYPE specified should be one of these values: warehouse / supplier.
+* Keywords specified are case-insensitive.
+* The TYPE specified should be one of these values: s/w.
+* Any combination of the name, product and remark prefix can be provided but at least one of the prefix for name,
+product and remark must be specified.
```
Comment (jeffreytjs) [inline docs/UserGuide.md]: Minor typo, should be "product or remark" instead

File: docs/UserGuide.md [hunk 2]
```diff
@@ -177,19 +177,21 @@ the warehouse at index 1 of the list of warehouses.
...
 
 Examples:
 
-* `find warehouse PANADOL SUSP` Displays all the warehouses managed by the manager that has a product named PANADOL SUSP.
-* `find supplier masks` Displays all the suppliers that have stock for the input product.
+* `find ct/s pd/masks` Displays all the suppliers that sell masks.
+* `find ct/w pd/PANADOL SUSP r/biggest` Displays the biggest warehouse that stores PANADOL SUSP.
```
Comment (jeffreytjs) [inline docs/UserGuide.md]: Should it display the biggest warehouse with any PANADOL product or SUSP product instead?

File: src/main/java/seedu/clinic/model/supplier/SupplierPredicate.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+ */
+public class SupplierPredicate implements Predicate<Supplier> {
+    private final Optional<List<String>> nameKeywords;
+    private final Optional<List<String>> productKeywords;
+    private final Optional<List<String>> remarkKeywords;
+
+    /**
+     * Constructs a new {@code SupplierPredicate}.
+     */
+    public SupplierPredicate(List<String> nameKeywords, List<String> productKeywords,
```
Comment (jeffreytjs) [inline src/main/java/seedu/clinic/model/supplier/SupplierPredicate.java:22]: Not sure if it's helpful but one way to make it more concise is to initialize all nameKeywords, productKeywords and remarkKeywords to Optional.empty() then only keep 3 if statement to update them if necessary

File: src/main/java/seedu/clinic/model/warehouse/WarehousePredicate.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    private final Optional<List<String>> nameKeywords;
+    private final Optional<List<String>> productKeywords;
+    private final Optional<List<String>> remarkKeywords;
+
+    /**
+     * Constructs a new {@code WarehousePredicate}.
+     */
+    public WarehousePredicate(List<String> nameKeywords, List<String> productKeywords,
+            List<String> remarkKeywords) {
+        if (nameKeywords.size() == 0) {
```
Comment (jeffreytjs) [inline src/main/java/seedu/clinic/model/warehouse/WarehousePredicate.java]: Same as above

File: src/test/java/seedu/clinic/logic/commands/FindCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+                Collections.singletonList("first warehouse"));
+        WarehousePredicate secondWarehousePredicate =
+                new WarehousePredicate(Collections.singletonList("second warehouse"),
+                Collections.singletonList("second warehouse"),
+                Collections.singletonList("second warehouse"));
+
+
+        FindCommand firstFindCommand = new FindCommand(firstSupplierPredicate);
+        FindCommand secondFindCommand = new FindCommand(secondSupplierPredicate);
+        FindCommand thirdFindCommand = new FindCommand(firstWarehousePredicate);
```
Comment (jeffreytjs) [inline src/test/java/seedu/clinic/logic/commands/FindCommandTest.java]: Can consider renaming the function name to contain Supplier and Warehouse for better readability

File: src/test/java/seedu/clinic/logic/commands/HelpCommandTest.java [hunk 6]
```diff
@@ -119,18 +118,19 @@ public void execute_exitHelp_success() {
...
-                + "KEYWORD specified is case-insensitive and must contain at least one keyword.\n"
-                + "The TYPE specified should be one of these values: warehouse / supplier.";
-        String findProductsInSupplierOrWarehouseSampleCommand = "Sample Command: \n"
-                + "find warehouse PANADOL SUSP \nfind supplier masks";
+        String findDescription = "Find\nFinds all suppliers or warehouses whose name, remark and/or"
+                + " products matches the argument keywords provided.";
+        String findCommandFormat = "Command format: \nfind ct/TYPE [n/NAME...]"
+                + " [pd/PRODUCT_NAME...] [r/REMARK...]\n\n"
+                + "TYPE should be one of these values: s / w.\n"
+                + "At least one of the name, product and remark prefix must be provided.\n"
```
Comment (jeffreytjs) [inline src/test/java/seedu/clinic/logic/commands/HelpCommandTest.java]: Should be "name, product or remark" instead

**Review by jeffreytjs — COMMENTED** [review]
Looks pretty good! Just a bit of typo found!

**Review by jeffreytjs — APPROVED** [review]
Approved!

**Review by Criss-Wang — APPROVED** [review]
LGTM!

**Pull request merged**